cmd/juju/charmhub/download.go

```diff
@@ -123,6 +130,10 @@ func (c *downloadCommand) Init(args []string) error {
 	}
 	// Allow for both <charm> and ch:<charm> to download.
 	c.charmOrBundle = curl.Name
+	c.revision = curl.Revision
+	if c.revision != -1 && c.channel != "" {
+		return errors.New("--channel cannot be specified together with a specific charm revision")
```

Comment:
Interesting that --channel is refused here while it is allowed for deploy (because post-deploy you might want to refresh, and we need to know what you are tracking). It is probably appropriate, though, as --channel would imply a revision to download.

However, I don't want to use `curl.Revision` for this as much as a separate CLI parameter.

In fact, `juju deploy` now refuses this syntax with:
```
$ juju deploy mysql-196 --channel 8.0/stable
ERROR cannot specify revision in a charm or bundle name. Please use --revision.
```